File: C/2020/12限/EarlyPeriod/if3.c
//   if3.c
//   cl  if3.c   get_line.c
#include <stdio.h>
#include <string.h>
#include "get_line.h"

#define BUFFER 128

int main() {
    char buffer[BUFFER];

    //
    get_line(buffer, BUFFER);
    printf("input is: %s \n", buffer);

    // 
    char *s = "ABC";
    //if ( s == buffer ) {
    if ( 0 == strcmp(buffer, s) ) {
        printf("input OK!! \n");
    } else {
        printf("%s is not %s \n", buffer, s);
    }

    return 0;
}

File: GameServer/2026/Sat34/if3.php
<?php  // if3.php

var_dump(true);
echo "<br>\n";
var_dump(false);
echo "<br>\n";

if (true) {
    echo "in true <br>\n";
}

if (1) {
    echo "in 1 <br>\n";
}

var_dump((bool)1);
echo "<br>\n";
var_dump((bool)0);
echo "<br>\n";
var_dump((bool)-1);
echo "<br>\n";

var_dump((bool)"abc");
echo "<br>\n";
var_dump((bool)"");
echo "<br>\n";
var_dump((bool)"0");
echo "<br>\n";
var_dump((bool)"00");
echo "<br>\n";

